### ISSUE TITLE
reload redis database on a rtpengine stby instance

### ISSUE THREAD

**Issue by descartin** [sipwise/rtpengine#660]
Hello all

we are making some HA tests with rtpengine 6.5.2.0+0~mr6.5.2.0 git-v6.5-e900fed

the idea should be having 2 servers (A and B) running a rtpengine instance each, with a same redis server where we can store the call information. We use floating IPs and at first IP would be active at server A, so we could switchover the IPs from server A to server B and the idea should be that the rtpengine instance on server B could continue the calls.

I can do that it seems but i have to restart rtpengine process when doing the IP switchover, otherwise, seems rtpengine doesnt not know about the callid of a call in an early dialog state.
I mean, if server A processes the offer, and after that i do a switchover, server B doesnt not know about that call to process the answer unless i restart the rtpengine on server B just after i do the IP switchover, so the rtpengine gets from the redis db all the call info and loads if to memory (¿?)

would there be any way to reload that without killing the rtpengine? i tried to set rediscmdtimeout to 0 

Nov 19 16:55:04 ip-10-100-10-109 rtpengine[10099]: INFO: Got CLI command:set rediscmdtimeout 0
Nov 19 16:55:04 ip-10-100-10-109 rtpengine[10099]: INFO: Asking Redis whether it's master or slave...
Nov 19 16:55:04 ip-10-100-10-109 rtpengine[10099]: INFO: Connected to Redis 10.100.10.70:6389 in master mode

but it didnt work either 

thanks a lot and regards
david

**Comment by rfuchs** [comment]
Yeah you can use the Redis key subscription feature. This has been developed by @smititelu and @inf265 so I can't personally comment on it any further.

**Comment by rfuchs** [comment]
Without using the key subscription feature, the idea is indeed to have the standby rtpengine not running and only have it start upon failover.

**Comment by descartin** [comment]
hello Richard

yes using the subscription feature worked for me, i was missing to enable notify-keyspace-events on redis configuration

thanks a lot and regards
david
